File: packages/core/src/task.ts
import { z } from "zod";

/**
 * A unit of work assigned to an agent or crew.
 */
export const TaskSchema = z.object({
  id: z.string(),
  description: z.string(),
  agentId: z.string().optional(),
  input: z.record(z.unknown()).optional(),
  output: z.unknown().optional(),
});
export type Task = z.infer<typeof TaskSchema>;

File: python/src/agent_framework/task.py
"""Task: unit of work assigned to an agent or crew."""

from pydantic import BaseModel
from typing import Any


class Task(BaseModel):
    """A unit of work."""

    id: str
    description: str
    agent_id: str | None = None
    input: dict[str, Any] | None = None
    output: Any = None
